xbmc/guilib/GUIWindowManager.cpp

```diff
@@ -810,6 +813,16 @@ void CGUIWindowManager::CloseDialogs(bool forceClose) const
   }
 }
 
+void CGUIWindowManager::CloseModalDialogs(bool forceClose) const
+{
+  CSingleLock lock(g_graphicsContext);
```

Comment:
`CGUIWindowManager::CloseDialogs` does the same. This looks like a copy with an additional condition. We have to protect against modifications of the `m_activeDialogs` vector.